libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo/network/http/HttpNetworkTransport.kt

```diff
@@ -392,4 +406,44 @@ private constructor(
       return chain.proceed(request.newBuilder().addHeaders(headers).build())
     }
   }
+
+  /**
+   * The protocol to use for incremental delivery (`@defer` and `@stream`).
+   */
+  sealed interface IncrementalDeliveryProtocol {
+    @ApolloInternal
+    val acceptHeader: String
+
+    @ApolloInternal
+    fun newIncrementalResultsMerger(): IncrementalResultsMerger
```

Comment:
What about a sealed marker interface to avoid exposing the interface functions? I also think it needs to be @ApolloExperimental?

```kotlin
@ApolloExperimental
sealed interface IncrementalResultsMerger
```

I don't think we want to allow users to implement their own `IncrementalResultsMerger` at this point?

Reply:
I went this route: 
- `IncrementalDeliveryProtocol` is a public enum
- `IncrementalDeliveryProtocolImpl` are internal `IncrementalResultsMerger` sealed interfaces

Reply:
Enum is even better 👍